### ISSUE TITLE
Automatic clicking "Enter VR" via javascript in unit test not working

### ISSUE THREAD

**Issue by vedhant** [stalgiag/p5.xr#34]
In the unit tests, I am automatically clicking the "Enter VR" button like this :
```javascript
let VRbutton = document.querySelector('header button');
VRbutton.click();
```
However this does not work. This is happening in the polyfilled version (I did not test for non-polyfill). More specifically, ```device.requestSession()``` in ```onXRButtonClicked()``` is not working and gives me a domexception error when I catch it. In fact, this also happens when I call the function in the src code itself and not in the test file (while normally serving build p5xr).

**Comment by stalgiag** [comment]
You might be able to simulate this with `dispatchEvent()`. I am not 100% sure how these event protections work but I think that this interaction has to come in the form of an interaction event.

**Comment by vedhant** [comment]
I tried dispatchEvent() and it did not work. It is the same as ```button.click()```. The problem which I suspect is the call site of ```device.requestSession()```. I am not able to figure out when this function works. If i call this in src code itself it does not work. If it is called within click event handler, it works(when trigerred by user and not simulated). If i call this in the chrome developer console it again works. I think it is somehow detecting whether the function is called by user intervention. I also tried calling this function after a 10s timeout in the src code and test file which failed and proves the above point. Im quite confused.
